osu.Game/Skinning/Skin.cs

```diff
@@ -57,7 +58,7 @@ protected Skin(SkinInfo skin, IStorageResourceProvider resources)
 
                 string jsonContent = Encoding.UTF8.GetString(bytes);
 
-                DrawableComponentInfo[skinnableTarget] = JsonConvert.DeserializeObject<IEnumerable<SkinnableInfo>>(jsonContent).ToArray();
+                DrawableComponentInfo[skinnableTarget] = JsonConvert.DeserializeObject<IEnumerable<SkinnableInfo>>(jsonContent).AsNonNull().ToArray();
```

Comment:
I don't think we want to crash as a result of null here (which is why i had the checks above).

Reply:
Fair point, tunnel-visioned a bit on that one. Figured that if a `null` falls out of json deserialisation it's a garbage-in-garbage-out scenario already, but I suppose the garbage can just as well be ignored.

Reply:
I think this should be better now with 69fc072, if I got the intention right.